jupyterhub_traefik_proxy/traefik_utils.py

```diff
@@ -0,0 +1,202 @@
+import time
+import socket
+import sys
+import requests
+import etcd3
+from os.path import abspath, dirname, join
+from subprocess import Popen
+from urllib.parse import urlparse
+
+
+_ports = {
+    "traefik": 8000,
+    "default_backend": 9000,
+    "first_backend": 9090,
+    "second_backend": 9099,
+}
```

Comment:
These ports probably belong in test code, since they shouldn't be part of configuration. Let's move the utilities for testing to a `tests/utils.py`to make it clear what's here for testing instead of for the normal runtime of the proxy.